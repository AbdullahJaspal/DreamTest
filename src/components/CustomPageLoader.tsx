import {Dimensions, StyleSheet, View} from 'react-native';
import React from 'react';
import FastImage from '@d11/react-native-fast-image';
import {gifs} from '../assets/gifs';

interface CustomPageLoaderProps {
  isLoading: boolean;
}

const {width, height} = Dimensions.get('screen');

const CustomPageLoader: React.FC<CustomPageLoaderProps> = ({isLoading}) => {
  return isLoading ? (
    <View style={styles.main_container}>
      <FastImage source={gifs.tiktokLoader} style={styles.loader} />
    </View>
  ) : null;
};

export default React.memo(CustomPageLoader);

const styles = StyleSheet.create({
  main_container: {
    width: width,
    height: height,
    zIndex: 100000,
    backgroundColor: 'transparent',
    alignItems: 'center',
    justifyContent: 'center',
  },
  loader: {
    width: 40,
    height: 40,
  },
});
