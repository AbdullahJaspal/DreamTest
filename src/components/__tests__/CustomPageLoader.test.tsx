import 'react-native';
import React from 'react';
import CustomPageLoader from '../CustomPageLoader';

import {it} from '@jest/globals';
import renderer from 'react-test-renderer';

it('CustomPageLoader renders correctly', () => {
  // Render the CText component
  const tree = renderer
    .create(<CustomPageLoader isLoading={true}></CustomPageLoader>)
    .toJSON();

  // Compare the rendered output to the saved snapshot
  expect(tree).toMatchSnapshot();
});
