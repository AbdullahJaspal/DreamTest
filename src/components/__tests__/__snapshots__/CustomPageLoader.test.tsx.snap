// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPageLoader renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
      "zIndex": 100000,
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "height": 40,
          "width": 40,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
